0003-interledger-protocol/0003-interledger-protocol.md

```diff
@@ -314,6 +314,7 @@ Relative errors indicate that the payment did not have enough of a margin in ter
 | **R00** | **Transfer Timed Out** | The transfer timed out, meaning the next party in the chain did not respond. This could be because you set your timeout too low or because something look longer than it should. The sender MAY try again with a higher expiry, but they SHOULD NOT do this indefinitely or a malicious connector could cause them to tie up their money for an unreasonably long time. |
 | **R01** | **Insufficient Source Amount** | Either the sender did not send enough money or the exchange rate changed before the payment was prepared. The sender MAY try again with a higher amount, but they SHOULD NOT do this indefinitely or a malicious connector could steal money from them. |
 | **R02** | **Insufficient Timeout** | The connector could not forward the payment, because the timeout was too low to subtract its safety margin. The sender MAY try again with a higher expiry, but they SHOULD NOT do this indefinitely or a malicious connector could cause them to tie up their money for an unreasonably long time. |
+| **R03** | **Superfluous Source Amount** | The transfer amount is sufficient to cover the costs of the payment, but this error is thrown as a courtesy, to warn that a lower transfer amount would also suffice. The sender SHOULD obtain a new quote and retry the Interledger payment with that adjusted transfer amount. |
```

Comment:
How much higher would it have to be for you to throw this error? Seems like it could be kind of annoying if you intentionally send a high buffer because you really want the payment to go through on the first try and some connector who is trying to do you a favor gives you this error.

Reply:
Any amount that does not match the amount from a recent quote will be wrong. It's up to the ILSP to decide what is "recent".

Examples:
1. As an end-user, I do a quote-by-destination for destination amount 123. It tells me source amount 15.
1. As an end-user, I do a quote-by-source for source amount 15. It tells me destination amount 123. I then do the payment with packet amount 123 and transfer amount 15.
1. As an end-user, I do a curve-quote. My ILSP give me a curve: `[14, 122], [16, 124]`.

In all cases, I wait for say 20 seconds (e.g. waiting for user to click 'OK') and then do the payment with packet amount 123 and transfer amount 15.

Say the market moves a little bit during these 20 seconds, but not too much. The ILSP will then e.g. have a grace period of say 60 seconds during which it will serve both old and new quote.

Say the market moves a lot during these 20 seconds. The ILSP will then reject the payment with either `InsufficientSourceAmount` or `SuperfluousSourceAmount`, to allow the end-user to re-quote. The ILSP should keep enough margin so that such re-quotes are only necessary in relatively rare circumstances.

Reply:
So if the market doesn't move and the user submits a source amount of 16, is that invalid? What if the curve the user got is a bit imprecise and they are giving a higher number because of a rounding error? What if the user wants to add an extra buffer to give the payment a higher chance of going through? 

What if there are more connectors between the sender and the connector in question and due to rounding, accident, or other connectors' rates fluctuating more money ends up at this connector than they wanted? Would every connector need to remember not only their own rates but also the exact rate they should use when passing a payment to a specific peer?